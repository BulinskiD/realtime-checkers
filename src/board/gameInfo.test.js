import React from 'react';
import GameInfo from './gameInfo';
import {shallow} from 'enzyme';

describe("GameInfo", () => {

    const modalRoot = global.document.createElement('div');
    modalRoot.setAttribute('id', 'root');
    const body = global.document.querySelector('body');
    body.appendChild(modalRoot);

    it('should match snapshot for given props', () => {
        expect(shallow(<GameInfo message={"Grają białe"} isEnded={false} />)).toMatchSnapshot();
    });

    it('should render modal on game end', () => {
        expect(shallow(<GameInfo message={"Wygrały białe"} isEnded={true} />)).toMatchSnapshot();
    });

});