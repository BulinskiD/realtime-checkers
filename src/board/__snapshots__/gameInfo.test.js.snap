// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameInfo should match snapshot for given props 1`] = `
<div>
  Grają białe
</div>
`;

exports[`GameInfo should render modal on game end 1`] = `
<Portal
  containerInfo={
    <div
      id="root"
    />
  }
>
  <InfoModal
    buttonText="Ok"
    onClose={[Function]}
    show={true}
    title="Gra zakończona"
  >
    Wygrały białe
  </InfoModal>
</Portal>
`;
